src/main/java/io/stargate/sgv2/jsonapi/service/rerank/operation/RerankProvider.java

```diff
@@ -0,0 +1,31 @@
+package io.stargate.sgv2.jsonapi.service.rerank.operation;
+
+import io.smallrye.mutiny.Uni;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class RerankProvider {
+  protected static final Logger logger = LoggerFactory.getLogger(RerankProvider.class);
+  protected final String baseUrl;
+  protected final String modelName;
+
+  protected RerankProvider() {
+    this(null, null);
+  }
+
+  protected RerankProvider(String baseUrl, String modelName) {
+    this.baseUrl = baseUrl;
+    this.modelName = modelName;
+  }
+
+  public abstract Uni<Response> rerank(int batchId, String query, List<String> passages);
+
+  public record Response(int batchId, List<Rank> ranks) {
+    public static Response of(int batchId, List<Rank> ranks) {
```

Comment:
-1 this of() is not needed at all, it is the same as constructor and should be removed